src/DatePicker/DatePickerInput.tsx

```diff
@@ -1,15 +1,24 @@
-import PropTypes from 'prop-types';
+import { FC } from 'react';
+import { DayPickerInputProps } from 'react-day-picker';
 import DayPickerInput from 'react-day-picker/DayPickerInput';
 
 import datePickerDefaults from './datePickerDefaults';
 import DatePickerInputOverlay from './DatePickerInputOverlay';
-import Input from '../Input';
+import Input, { InputProps } from '../Input';
 
 const defaultInputProps = {
   autoComplete: 'off'
 };
 
-const DatePickerInput = ({ dayPickerProps, inputProps, ...props }) => (
+type Props = Omit<DayPickerInputProps, 'inputProps'> & {
+  inputProps?: InputProps;
+};
```

Comment:
`inputProps` is typed as `object` in `DayPickerInputProps`, but we can type it more accurately here since we're passing our `Input` as the component below.